pkg/endpoint/restore.go

```diff
@@ -304,8 +304,11 @@ func (e *Endpoint) restoreIdentity(regenerator *Regenerator) error {
 				Group: initialGlobalIdentitiesControllerGroup,
 				DoFunc: func(ctx context.Context) (err error) {
 					err = e.allocator.WaitForInitialGlobalIdentities(ctx)
-					if err != nil {
+					switch {
+					case !errors.Is(err, context.Canceled):
```

Comment:
I was assuming that when this context is canceled, the endpoint's context is also canceled.. but given the CI results I'm second-guessing whether that's true.

Reply:
I think the key thing here is the call to `e.UpdateController`: https://github.com/cilium/cilium/blob/f28330123edacbcac9d8503a33564cc41e62cacc/pkg/endpoint/endpoint.go#L534. "This" context here is the controller context which is set to the endpoint's context.

Reply:
I wonder why some regenerations appear to be hanging "forever" in CI then :thinking:

Reply:
Ah, I think this is due to a simple mistake of not tightening the `err` check: https://github.com/cilium/cilium/compare/6d4893ca5acb05d2dc3dd5a727f361d9e8e35ac8..d0f8767b5b06a2aa493610de969105f626a5e5e2

That should do it. Without this, even non-nil errors were going through the `fallthrough` and ending up at a premature `return`.